webhook/core/annotation_validation_webhook.go

```diff
@@ -84,6 +84,31 @@ func (av *AnnotationValidator) Handle(ctx context.Context, req admission.Request
 			}
 		}
 	}
+
+	// check if pod-sg annotation is updated by any resource
+	if podSgAnnot, ok := pod.Annotations["fargate.amazonaws.com/pod-sg"]; ok {
+		webhookLog.V(1).Info("Got annotation:", "fargate.amazonaws.com/pod-sg", podSgAnnot)
+		if req.Operation == v1beta1.Update {
+			webhookLog.V(1).Info("Operation is update, checking that pod-sg annotation wasn't modified")
+			oldPod, err := av.K8sWrapper.GetPodFromAPIServer(pod.Namespace, pod.Name)
```

Comment:
This adds a call to the API Server directly instead of getting the object from cache. Is there a reason why you wouldn't want to get the pod from cache instead?

Could you please also add the latency introduced by these changes under load. Example before this changes it was taking x milliseconds and after this change it was talking x+n milliseconds. Once we have the number n, we can be more confident the change is not introducing a large latency on updates.

We can sync up offline on how we can test this.